### ISSUE TITLE
screen.workarea is not updated for client struts when switching tags

### ISSUE THREAD

**Issue by coldfix** [awesomeWM/awesome#1434]
Output of `awesome --version`:

```
% awesome --version
awesome v4.0 (Harder, Better, Faster, Stronger)
 • Compiled against Lua 5.3.3 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • RandR 1.5 support: ✔
 • LGI version: 0.9.1
```

**How to reproduce the issue:**

In the default `rc.lua` add a rule like this:

```lua
    { rule = { class = "Pidgin", role = "buddy_list" },
        properties = {
          floating=true,
          maximized_vertical=true,
          maximized_horizontal=false,
          tag="2",
          switchtotag=false,
        },
        callback = function (c)
            c:struts({left=350, right=0, top=0, bottom=0})
            -- not taking into account screen.workarea.x for demonstrational purposes:
            c:geometry({x=0, width=350})
        end },
```

**Actual result:**

`screen.workarea` is updated only once, leading to different behaviour depending on whether the client is initially visible:
- if the client is created on a non-active tag, no space is reserved for the client.
- if the client is on the active tag, space is reserved on the left on all tags, leading to empty space for every tag where the client is not shown.

**Expected result:**

- `screen.workarea` is updated on each switch of tag, i.e. 
- space is reserved on the left on all tags where the client is visible - and only on those.
- when later making the client visible on another tag, space should be reserved on the respective tag.

Worked as expected in awesome 3.5.

**Comment by coldfix** [comment]
Also note that
- you need a tiling layout to observe these effects.
- this also effects docked clients

**Comment by Elv13** [comment]
Looks like https://github.com/awesomeWM/awesome/pull/1342, can you test with git-master?

**Comment by coldfix** [comment]
Indeed, thank you. Sorry, I searched the github issue, but didn't pay attention that github by default restricts to open issues.

**Comment by psychon** [comment]
This might be the second duplicate of this bug report already (I'm not sure and too lazy to look it up). This is the wrong place to discuss it, but: Might be already time for a bugfix release soon-ish? Which things do we want to be fixed before? #1418 looks quite important to me and is well enough understood that one (I?) could try to fix it this week.

**Comment by Elv13** [comment]
> This is the wrong place to discuss it,

I asked on IRC, but I am not sure if your reallt there or just the proxy, so

```
[06:17] <Elv13> I am ok with a release.
[06:18] <Elv13> The deadline for Debian is Jan 25, 4.0 is merged into Sid (I harassed (according to him) jd and arnaud until they did it). It should be auto-merged into Stretch in 6 days
[06:19] <Elv13> So it would be playing with fire to push a new one now, but possible
[06:19] <Elv13> Also, can we call it 4.1 after all or do you really want to keep 3 numbers?
```
